EFCore/MAUI/Readme.md

```diff
@@ -0,0 +1,873 @@
+      public async Task<byte[]> GetAuthorPhotoAsync(Guid postId)
+          => await HttpClient.GetByteArrayAsync($"{_apiUrl}CustomEndPoint/AuthorPhoto/{postId}");
+   }
+   ```
+
+3. Implement logic required to display the detail view when a user taps a post in the list.
+
+   _ViewModels/ItemDetailViewModel.cs_:
+
+   ```cs
+   public ImageSource Thumbnail 
+   => ImageSource.FromStream(() => new MemoryStream(_photoBytes));
+
+   public async Task LoadItemId(string itemId) {
+      try {
+         _photoBytes = await DataStore.GetAuthorPhotoAsync(Guid.Parse(itemId));
+         OnPropertyChanged(nameof(Thumbnail));
+         Post = await DataStore.GetItemAsync(itemId);
+         Id = Post.ID;
+         Title = Post.Title;
+         Content = Post.Content;
+         
+      }
+      catch (Exception e) {
+         System.Diagnostics.Debug.WriteLine($"Failed to Load Post {e}");
+      }
+   }
+   ```
+
+   _ViewModels/ItemsViewModel.cs_:
+
+   ```cs
+    public class ItemsViewModel : BaseViewModel {
+        Post _selectedPost;
+
+        public ItemsViewModel() {
+            // ...
+            ItemTapped = new Command<Post>(OnItemSelected);
+        }
+
+        public Command<Post> ItemTapped { get; }
+
+        public Post SelectedPost {
+            get => _selectedPost;
+            set {
+                SetProperty(ref _selectedPost, value);
+                OnItemSelected(value);
+            }
+        }
+
+        async void OnItemSelected(Post post) {
+            if (post != null) await Navigation.NavigateToAsync<ItemDetailViewModel>(post.ID);
+        }
+        // ...
+    }
+   ```
+
+   _App.xaml.cs_:
+
+   ```cs
+   public App() {
+      // ...
+      Routing.RegisterRoute(typeof(ItemDetailPage).FullName, typeof(ItemDetailPage));
+      // ...
+   }
+   ```
+   
+   _Views/ItemDetailPage.xaml_:
+
+   ```xml
+   <StackLayout Spacing="5" Padding="15">
+      <Label Text="Title:" FontFamily="Roboto" FontSize="12" TextColor="{StaticResource NormalLightText}"/>
+      <Label Text="{Binding Title}" FontFamily="Roboto" FontSize="14" TextColor="{StaticResource NormalText}" Margin="0, 0, 0, 15"/>
+      <Label Text="Content:" FontFamily="Roboto" FontSize="12" TextColor="{StaticResource NormalLightText}" />
+      <Label Text="{Binding Content}" FontFamily="Roboto" FontSize="14" TextColor="{StaticResource NormalText}"/>
+      <Label Text="Author:"></Label>
+      <Image   Source="{Binding Thumbnail}"></Image>
+   </StackLayout>
+   ```
+
+![](../../images/MAUI/DetailView.png)
+
+### Add a "New Item" entry form
+
+1. The Web API Service automatically generates OData endpoints required to create business objects. However, because the example application's security system is configured to disallow post creation for some users, it is beneficial to have a way to check a user's permissions before they can try to submit a post. To do this, implement a custom `CanCreate` endpoint.
```

Comment:
Two main reasons why I started an edit: 

- long sentence
- the phrase "it is beneficial to have" (weak verbs)

```suggestion
1. The Web API Service automatically generates OData endpoints required to create business objects. However, the example application's security system is configured to disallow post creation for some users. You want to check permissions before a user can try and submit a post. For this purpose, implement a custom `CanCreate` endpoint.
```